### PR TITLE
net-misc/i2pd: version bump 2.6.0

### PR DESCRIPTION
**Pull request by khumarahn**
This is an update. I had to make some small changes in initd, confd, service files, and add a patch https://github.com/PurpleI2P/i2pd/commit/5c877de2c2f954c27471092fff73443a1c9af9d4 without which websites would not open on a fresh install (with empty /var/lib/i2pd/addressbook).

I tested the new ebuild, and it seems working. It would not hurt if somebody tests it too.

The new version brings a graceful shutdown on SIGINT, and I do not know if (and how) we should account for this in service files. Graceful shutdown takes 10 minutes, because this is what takes for tunnel commitments to expire.

### CODE REVIEW

**Comment by khumarahn** [comment]
https://bugs.gentoo.org/show_bug.cgi?id=578770

File: net-misc/i2pd/files/i2pd-2.6.0-disable_ipv6_in_i2pd_conf.patch [hunk 1]
```diff
@@ -0,0 +1,9 @@
+diff -Naur i2pd-2.5.1/debian/i2pd.conf i2pd-2.5.1b/debian/i2pd.conf
+--- i2pd-2.5.1/debian/i2pd.conf	2016-03-10 19:34:16.000000000 +0100
++++ i2pd-2.5.1b/debian/i2pd.conf	2016-03-24 22:30:51.490011172 +0100
+@@ -1,4 +1,4 @@
+-ipv6
++ipv6 = false
+
```
Comment (blueness) [inline net-misc/i2pd/files/i2pd-2.6.0-disable_ipv6_in_i2pd_conf.patch]: Why this change in syntax?

File: net-misc/i2pd/files/i2pd-2.6.0-fix_installed_components.patch [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+@@ -360,10 +360,6 @@
+   endif ()
+ endif ()
+ 
+-install(FILES ../LICENSE
+-  DESTINATION .
+-  COMPONENT Runtime
+-  )
+ # Take a copy on Appveyor
+ install(FILES "C:/projects/openssl-$ENV{OPENSSL}/LICENSE"
```
Comment (blueness) [inline net-misc/i2pd/files/i2pd-2.6.0-fix_installed_components.patch]: We don't want to downloading a file during the compile because it can change upstream and its not recorded in the manifest.

File: net-misc/i2pd/files/i2pd-2.6.0-fix_installed_components.patch [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ #   COMPONENT Source FILES_MATCHING
+@@ -382,7 +377,7 @@
+ #   )
+ 
+ file(GLOB I2PD_HEADERS "../*.h")
+-install(FILES ${I2PD_HEADERS} DESTINATION src/ COMPONENT Headers)
++install(FILES ${I2PD_HEADERS} DESTINATION "include/${PROJECT_NAME}" COMPONENT Headers)
+ # install(DIRECTORY ../ DESTINATION src/
+ #   # OPTIONAL
+ #   COMPONENT Headers FILES_MATCHING
```
Comment (blueness) [inline net-misc/i2pd/files/i2pd-2.6.0-fix_installed_components.patch]: I don't want to deviate from upstream's CMakeLists.txt.

File: net-misc/i2pd/files/i2pd-2.6.0-incomplete-hosts.patch [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
++++ b/AddressBook.h
+@@ -66,7 +66,7 @@ namespace client
+ 			void InsertAddress (const std::string& address, const std::string& base64); // for jump service
+ 			void InsertAddress (std::shared_ptr<const i2p::data::IdentityEx> address);
+ 
+-			void LoadHostsFromStream (std::istream& f);
++			bool LoadHostsFromStream (std::istream& f);
+ 			void DownloadComplete (bool success, const i2p::data::IdentHash& subscription, const std::string& etag, const std::string& lastModified);
+ 			//This method returns the ".b32.i2p" address
+ 			std::string ToAddress(const i2p::data::IdentHash& ident) { return GetB32Address(ident); }
```
Comment (blueness) [inline net-misc/i2pd/files/i2pd-2.6.0-incomplete-hosts.patch]: Again, why are we deviating from upstream?

File: net-misc/i2pd/files/i2pd-2.6.0.confd [hunk 5]
```diff
@@ -0,0 +1,9 @@
+I2PD_USER="${I2PD_USER:-i2pd}"
+I2PD_GROUP="${I2PD_GROUP:-i2pd}"
+I2PD_LOG="/var/log/i2pd.log"
+I2PD_PID="/var/run/i2pd.pid"
+I2PD_CFGDIR="/etc/i2pd/"
+# Options to i2pd
+I2PDOPTIONS="--daemon --service --pidfile=${I2PD_PID} \
+--log=file --logfile=${I2PD_LOG} \
+--conf=${I2PD_CFGDIR}i2pd.conf --tunconf=${I2PD_CFGDIR}tunnels.conf"
```
Comment (blueness) [inline net-misc/i2pd/files/i2pd-2.6.0.confd]: what's wrong with i2pd-2.5.1.confd?

File: net-misc/i2pd/files/i2pd-2.6.0.initd [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        sleep 1
+        [ -e "$I2PD_PID" -a -e /proc/$(cat "$I2PD_PID") ]
+        eend $?
+}
+
+stop() {
+        ebegin "Stopping ${SVCNAME}"
+        start-stop-daemon -K -p "${I2PD_PID}" -R SIGTERM/20 SIGKILL/20 -P
+        eend $?
+}
```
Comment (blueness) [inline net-misc/i2pd/files/i2pd-2.6.0.initd]: ditto.

**Comment by blueness** [comment]
I don't get why you're making all these changes over what's in the tree now.

**Comment by blueness** [comment]
I bump to 2.6.0, do you wan to propose changes against what i put on the tree pleaes.

**Comment by tomboy-64** [comment]
i will examine it and work the changes in, if appropriate.

File: net-misc/i2pd/files/i2pd-2.6.0-fix_installed_components.patch (hunk 2 above)
Comment (khumarahn) [inline net-misc/i2pd/files/i2pd-2.6.0-fix_installed_components.patch]: Hi, I do not understand which file we are downloading during compile phase.

File: net-misc/i2pd/files/i2pd-2.6.0-fix_installed_components.patch (hunk 3 above)
Comment (khumarahn) [inline net-misc/i2pd/files/i2pd-2.6.0-fix_installed_components.patch]: I would not deviate from upstream. Changes to CMakeLists are fully inherited from previous ebuilds. It looks like this patch strips out installation of i2pd sources and license file, and corrects path for installation of headers.

File: net-misc/i2pd/files/i2pd-2.6.0-incomplete-hosts.patch (hunk 4 above)
Comment (khumarahn) [inline net-misc/i2pd/files/i2pd-2.6.0-incomplete-hosts.patch]: This is a backport from mainstream: https://github.com/PurpleI2P/i2pd/commit/5c877de2c2f954c27471092fff73443a1c9af9d4 which was done after the release of 2.6.0. It fixes a problem with addressbook, where addressbook was not functioning on fresh installations, and no websites could be visited.

File: net-misc/i2pd/files/i2pd-2.6.0.confd (hunk 5 above)
Comment (khumarahn) [inline net-misc/i2pd/files/i2pd-2.6.0.confd]: In the mainstream, tunnels.cfg changed to tunnels.conf, nothing else. We are catching up.

File: net-misc/i2pd/files/i2pd-2.6.0.initd (hunk 6 above)
Comment (khumarahn) [inline net-misc/i2pd/files/i2pd-2.6.0.initd]: Here I removed -p option from start-stop-daemon, because the pid file is created by the daemon itself, and we check explicitly if it was created. Besides, with the -p option the service status after start was "crashed", probably because of the conflict in pid file creation.

**Comment by khumarahn** [comment]
I will post pull requests wrt current tree. Closing this pull request.

**Comment by khumarahn** [comment]
new pull request, more detailed, based on the current tree:
https://bugs.gentoo.org/show_bug.cgi?id=578770
